### PR TITLE
WIP: replaced Vec with HashSet in gstreamer muteables

### PR DESCRIPTION
**Pull request by khodzha**
@ferjm you've been suggesting HashSet from the start yet because we didn't have an id back then I went with Vec.

But since we already add usize ids to muteables Vec I suppose we can just utilize it for Hash + Eq and replace Vec with HashSet now.

(I'm not sure whether IdMuteable is appropriate naming at all and also whether it should go straight to `lib.rs` file or be separated into another one)

### CODE REVIEW

File: backends/gstreamer/lib.rs [hunk 1]
```diff
@@ -63,28 +63,50 @@ use servo_media_streams::registry::MediaStreamId;
...
 use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
 use std::sync::{Arc, Mutex, Weak};
-use std::vec::Vec;
 
 lazy_static! {
     static ref BACKEND_BASE_TIME: gst::ClockTime = { gst::SystemClock::obtain().get_time() };
 }
 
+struct IdMuteable {
+    id: usize,
```
Comment (ferjm) [inline backends/gstreamer/lib.rs]: `Player`s and `AudioContext`s already have their own ids. I think there is no need to duplicated them here.

File: backends/gstreamer/lib.rs [hunk 2]
```diff
@@ -63,28 +63,50 @@ use servo_media_streams::registry::MediaStreamId;
...
 use std::path::PathBuf;
 use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
 use std::sync::{Arc, Mutex, Weak};
-use std::vec::Vec;
 
 lazy_static! {
     static ref BACKEND_BASE_TIME: gst::ClockTime = { gst::SystemClock::obtain().get_time() };
 }
 
+struct IdMuteable {
```
Comment (ferjm) [inline backends/gstreamer/lib.rs]: nit: Let's rename this to `MuteableRef`.

File: backends/gstreamer/lib.rs (hunk 1 above)
Comment (khodzha) [reply backends/gstreamer/lib.rs]: yes, but we have only `Weak<Mutex<dyn Muteable>>`, is locking mutex just to get an id fine?

Comment (ferjm) [reply backends/gstreamer/lib.rs]: Hmm, I think that should be fine.

Comment (khodzha) [reply backends/gstreamer/lib.rs]: ah, this id is also used for Eq + Hash!

**Comment by khodzha** [comment]
After a discussion with @ferjm it was decided HashSet doesnt provide much of a benefit since in current implementation `retain()` loop cant be replaced with a `remove()` (which was the idea behind using HashSet in the first place).

So I'm closing this PR.

**Pull request closed without merging**